chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -307,31 +307,45 @@ func (n *NeutrinoNotifier) notificationDispatcher() {
 				currentHeight := n.bestHeight
 				n.heightMtx.RUnlock()
 
-				// Lookup whether the transaction is already included in the
-				// active chain.
-				txConf, err := n.historicalConfDetails(msg.TxID, currentHeight,
-					msg.heightHint)
-				if err != nil {
-					chainntnfs.Log.Error(err)
-				}
+				// Look up whether the transaction is already
+				// included in the active chain. We'll do this
+				// in a goroutine to prevent blocking
+				// potentially long rescans.
+				go func() {
+					confDetails, err := n.historicalConfDetails(
+						msg.TxID, currentHeight,
+						msg.heightHint,
+					)
+					if err != nil {
+						chainntnfs.Log.Error(err)
+					}
+
+					if confDetails != nil {
+						err := n.txConfNotifier.UpdateConfDetails(
+							*msg.TxID, msg.ID,
+							confDetails,
+						)
+						if err != nil {
+							chainntnfs.Log.Error(err)
+						}
+						return
+					}
 
-				if txConf == nil {
-					// If we can't fully dispatch confirmation,
-					// then we'll update our filter so we can be
-					// notified of its future initial confirmation.
+					// If we can't fully dispatch
+					// confirmation, then we'll update our
+					// filter so we can be notified of its
+					// future initial confirmation.
 					rescanUpdate := []neutrino.UpdateOption{
 						neutrino.AddTxIDs(*msg.TxID),
 						neutrino.Rewind(currentHeight),
 					}
-					if err := n.chainView.Update(rescanUpdate...); err != nil {
-						chainntnfs.Log.Errorf("unable to update rescan: %v", err)
+					err = n.chainView.Update(rescanUpdate...)
```

Comment:
What do you mean by this @halseth ?

Reply:
We are passing in `currentHeight`, which might be outdated at this point (because of potential long-running `historicalConfDetails`). I think this should be okay, but just meant that we should check that it indeed is.

Reply:
Good point, it's likely we'd rescan all of the manual dispatch done before? Since current height is behind a mutex, should be safe to read again?

Reply:
`currentHeight` is a copy so it should remain unchanged even if the actual height changes.